src/Gui/Inventor/Draggers/Gizmo.cpp

```diff
@@ -0,0 +1,697 @@
+{
+    setProperty(property);
+}
+
+SoInteractionKit* LinearGizmo::initDragger()
+{
+    draggerContainer = new SoLinearDraggerContainer;
+    draggerContainer->color.setValue(1, 0, 0);
+    dragger = draggerContainer->getDragger();
+
+    dragger->addStartCallback(
+        [] (void* data, SoDragger*) {
+            assert(data);
+            static_cast<LinearGizmo*>(data)->draggingStarted();
+        },
+        this
+    );
+    dragger->addFinishCallback(
+        [] (void* data, SoDragger*) {
+            assert(data);
+            static_cast<LinearGizmo*>(data)->draggingFinished();
+        },
+        this
+    );
+    dragger->addMotionCallback(
+        [] (void* data, SoDragger*) {
+            assert(data);
+            static_cast<LinearGizmo*>(data)->draggingContinued();
+        },
+        this
+    );
+
+    dragger->labelVisible = false;
+
+    setDragLength(property->value().getValue());
+
+    dragger->instantiateBaseGeometry();
+
+    // change the dragger dimensions
+    auto arrow = SO_GET_PART(dragger, "arrow", SoArrowGeometry);
+    arrow->cylinderHeight = 3.5;
+    arrow->cylinderRadius = 0.2;
+
+    return draggerContainer;
+}
+
+void LinearGizmo::uninitDragger()
+{
+    dragger = nullptr;
+    draggerContainer = nullptr;
+}
+
+GizmoPlacement LinearGizmo::getDraggerPlacement()
+{
+    assert(draggerContainer);
+    return {draggerContainer->translation.getValue(), draggerContainer->getPointerDirection()};
+}
+
+void LinearGizmo::setDraggerPlacement(const SbVec3f& pos, const SbVec3f& dir)
+{
+    assert(draggerContainer);
+    draggerContainer->translation = pos;
+    draggerContainer->setPointerDirection(dir);
+}
+
+void LinearGizmo::setDraggerPlacement(Base::Placement placement)
+{
+    assert(draggerContainer);
+    draggerContainer->translation = Base::convertTo<SbVec3f>(placement.getPosition());
+    draggerContainer->rotation = Base::convertTo<SbRotation>(placement.getRotation());
+}
+
+void LinearGizmo::reverseDir() {
+    auto dir = getDraggerContainer()->getPointerDirection();
+    getDraggerContainer()->setPointerDirection(dir * -1);
+}
+
+
+double LinearGizmo::getDragLength()
+{
+    double dragLength = dragger->translationIncrementCount.getValue()
+        * dragger->translationIncrement.getValue();
+
+    return (dragLength - addFactor) / multFactor;
+}
+
+void LinearGizmo::setDragLength(double dragLength)
+{
+    dragLength = dragLength * multFactor + addFactor;
+    dragger->translation = {0, static_cast<float>(dragLength), 0};
+}
+
+void LinearGizmo::setGeometryScale(float scale)
+{
+    dragger->geometryScale = SbVec3f(scale, scale, scale);
+}
+
+SoLinearDraggerContainer* LinearGizmo::getDraggerContainer()
+{
+    assert(draggerContainer);
+    return draggerContainer;
+}
+
+void LinearGizmo::setProperty(QuantitySpinBox* property)
+{
+    if (connection) {
+        QuantitySpinBox::disconnect(connection);
+    }
+
+    this->property = property;
+    connection = QuantitySpinBox::connect(
```

Comment:
Should be named `quantityChangedConnection` so it is clear what it represents.

Reply:
Should be fixed.